turbopack/crates/turbo-tasks-testing/tests/random_change.rs

```diff
@@ -0,0 +1,71 @@
+#![feature(arbitrary_self_types)]
+#![feature(arbitrary_self_types_pointers)]
+#![allow(clippy::needless_return)] // tokio macro-generated code doesn't respect this
+
+use anyhow::{bail, Result};
+use rand::Rng;
+use turbo_tasks::{State, Vc};
+use turbo_tasks_testing::{register, run, Registration};
+
+static REGISTRATION: Registration = register!();
+
+#[tokio::test]
+async fn random_change() {
+    run(&REGISTRATION, || async {
+        let state = make_state();
+        let value = rand::thread_rng().gen_range(0..100);
```

Comment:
I know `run` executes the closure multiple times (maybe there should be a comment about that here), but is there some way we could run this even more times than usual here?

I'd like it if this test could be made a bit more deterministic (e.g. using a shared `Arc<Mutex<_>>` seeded RNG constructed outside of the closure). Non-deterministic tests aren't as good because they'll sometimes randomly pass with retries.

Reply:
I can change the test to use a atomic counter, so it's deterministic...